src/Components/NewPerson/NewPerson.tsx

```diff
@@ -0,0 +1,208 @@
+import React, {
+  FormEvent, useContext, useEffect, useState,
+} from 'react';
+import './NewPerson.scss';
+import { useNavigate } from 'react-router-dom';
+import { PeopleContext } from '../PeopleContext';
+import { Person } from '../../Types/Person';
+
+export const NewPerson: React.FC = React.memo(() => {
+  const { people, setPeople } = useContext(PeopleContext);
+  const [name, setName] = useState('');
+  const [sex, setSex] = useState('');
+  const [born, setBorn] = useState(0);
+  const [died, setDied] = useState(0);
+  const [fatherName, setFatherName] = useState('');
+  const [motherName, setMotherName] = useState('');
```

Comment:
Let's create one state with an object instead of 5 ones. It' hard to manage 5 states